File: assessmentApp/src/app/services/book.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Book } from '../interfaces/Book';

@Injectable({
  providedIn: 'root'
})
export class BookService {

  private readonly apiUrl: string = 'http://localhost:5002/api/books';

  constructor(private http: HttpClient) { }

  getAllBooks(): Observable<any> {
    return this.http.get(`${this.apiUrl}`);
  }

  createBook(book: Book): Observable<any> {
    return this.http.post(`${this.apiUrl}`, book);
  }

  updateBook(id: number, updatedBook: Book): Observable<any> {
    return this.http.put(`${this.apiUrl}/${id}`, updatedBook);
  }

  deleteBook(id: number): Observable<any> {
    return this.http.delete(`${this.apiUrl}/${id}`);
  }

}

File: student/src/main/java/com/newproject/app/student/services/BookService.java
package com.newproject.app.student.services;

import java.util.List;

import com.newproject.app.student.models.Book;

public interface BookService {
    
    List<Book> getAllBooks();

    Book getBookById(Long id);

    Book createBook(Book book);

    Book updateBook(Long id, Book updatedBook);

    void deleteBook(Long id);
}
